src/layouts/Workspace/Workspace.tsx

```diff
@@ -19,170 +18,103 @@ import {
 
 import { ProtectedRouteWithProps } from "routing/ProtectedRouteWithProps"
 
-import {
-  Section,
-  SectionHeader,
-  SectionCaption,
-  CreateButton,
-} from "../components"
+import { PageData } from "types/directory"
+
+import { Section } from "../components"
 import { SiteViewLayout } from "../layouts"
 
-import {
-  ContactCard,
-  PageCard,
-  NavigationCard,
-  FolderCard,
-  HomepageCard,
-} from "./components"
+import { ContactCard, NavigationCard, HomepageCard } from "./components"
+import { FoldersAndPagesController } from "./FoldersAndPagesController"
 
 const CONTACT_US_TEMPLATE_LAYOUT = "contact_us"
 
-const WorkspacePage = (): JSX.Element => {
-  const {
-    params: { siteName },
-  } = useRouteMatch<{ siteName: string }>()
-
-  const { setRedirectToPage } = useRedirectHook()
-  const { url } = useRouteMatch()
-  const { data: _dirsData, isLoading: isDirLoading } = useGetFolders({
+// refers to the homepage, navigation bar and contact us pages.
+export const MainPages = (props: {
+  siteName: string
+  pagesData: PageData[]
```

Comment:
we don't seem to be using this except as an indication of whether the page is loading. perhaps we could change this prop to `isLoading`?

Reply:
hmm `MainPages` here refer to the top three main pages (homepage, navigation, and contact us pages) that exist in all pages by default, not as a placeholder while the pages are loading!

Reply:
i think you might've gotten confused - i'm not worried about the naming, more about the prop `pagesData`, which is only used as an indicator of whether it's loading. this should just be a boolean and called `isLoading` so that it's clearer and more indicative of what this component requires